main.go

```diff
@@ -191,6 +193,21 @@ func main() {
 		log.Fatal(err)
 	}
 
+	// Creates the output upload exclusions file, required by the JobCompose InitFromJob method.
+	excludeFile, err := os.Create(dcompose.UploadExcludesFilename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer excludeFile.Close()
+
+	excludeFileContents := strings.Join(job.ExcludeArguments(), "\n")
+
+	_, err = io.Copy(excludeFile, strings.NewReader(excludeFileContents))
```

Comment:
This is a minor thing, but might it be nice to not bother with allocating the fully put-together string at all? I'd imagine it should be possible to replace this stuff with a loop over `job.ExcludeArguments()` which in each loop does an `excludeFile.WriteString(arg)` + `excludeFile.WriteString("\n")` or so. I suppose it's not like the list of paths is likely to get so huge it's a major issue to keep `excludeFileContents` in memory, but why bother when it's unnecessary, right? :grimacing:

Reply:
Heh, I guess I'm too used to UI code now, where the `join` would be preferred, but now that you mention it, I think we always want a newline at the end of the file too, so the "exploded" loop might be the way to go 👍